app/models/request/MessageRequest.scala

```diff
@@ -23,5 +23,5 @@ import play.api.mvc.WrappedRequest
 
 abstract class BaseMessageRequest[A](request: Request[A], messageType: MessageType, instructions: GenInstructions) extends WrappedRequest[A](request)
 
-case class MessageRequest[A](request: Request[A], messageType: MessageType, instructions: GenInstructions)
+case class MessageRequest[A](request: ChannelRequest[A], messageType: MessageType, instructions: GenInstructions)
```

Comment:
Since we never use these two requests independently, can you change BaseMessageRequest[A] to extend ChannelRequest so that we can avoid making calls to x.request.channel?